tests/optim/test_oss.py

```diff
@@ -534,12 +534,13 @@ def closure():
 
 
 @skip_if_single_gpu
-def test_reproducibility():
+@pytest.mark.parametrize("broadcast_fp16", [False, True])
+def test_reproducibility(broadcast_fp16: bool):
     world_size = 2
     temp_file_name = tempfile.mkstemp()[1]
 
     mp.spawn(
-        run_test_reproducibility, args=(world_size, temp_file_name), nprocs=world_size, join=True,
+        run_test_reproducibility, args=(world_size, temp_file_name, broadcast_fp16), nprocs=world_size, join=True,
```

Comment:
Shall we add a benchmark for this? WDYT?

Reply:
ideally it would be on fb cluster, it only really makes sense when inter node

Reply:
from offline discussion since I was not clear why this only makes sense for inter node: Turns our the perf gain is insignificant due to fast intranode comms. We still want to measure performance to make sure there are no regressions.

Reply:
I can add that in another PR if that's ok ? I tested locally and not a big difference, if anything a bit slower with this option turned on, as expected